HTTPRequest/HTTPRequest.Unity.cs

```diff
@@ -0,0 +1,146 @@
+﻿// IF WE ARE USING UNITY, USE THIS VERSION OF THE CLASS
+#if UNITY_5
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Text;
+using System.Reflection;
+
+using UnityEngine;
+
+
+public class HTTPRequest {
+	private WWW request;
+	private CookieContainer cookies;
+	private Action<Exception, string> cb;
+	private Stopwatch timeoutTimer = new Stopwatch();
+
+	//
+	public double timeout = 100 * 1000;
+
+
+	//
+	public HTTPRequest(string url, string contentType, byte[] postData, Dictionary<string, string> headers, CookieContainer cookies, Action<Exception, string> cb) {
+		// Start timeout timer
+		timeoutTimer.Start();
+
+		// Queue constructor for main thread execution
+		HTTPRequestManager.Queue(Constructor(url, contentType, postData, headers, cookies, cb));
+	}
+
+
+	//
+	private IEnumerator Constructor(string url, string contentType, byte[] postData, Dictionary<string, string> headers, CookieContainer cookies, Action<Exception, string> cb) {
+		Dictionary<string, string> headersCopy = new Dictionary<string, string>(headers);
```

Comment:
Seems like `headers` can be null in some cases (at least on the Gundam frontend) and Dictionary will complain about it being null resulting in some nasty warnings in the console.